stac_pydantic/api/utils/link_factory.py

```diff
@@ -0,0 +1,94 @@
+import inspect
+from dataclasses import dataclass
```

Comment:
Add `dataclasses` requirement to `install_requires` for Python 3.6?

Reply:
nice catch